StRoot/StBTofUtil/StBTofSimResParams.h

```diff
@@ -20,80 +21,99 @@
+			return params[ itray ][ imodule * 6 + icell ];
+		}
 
-		TDataSet *DB = GetDataBase("Calibrations/tof/tofSimResParams");
+		//! Loads BTOF Sim Params from database
+		//void loadParams(const int date = 20160913, const int time = 175725, const char* Default_time = "2016-09-13 17:57:25")
+		void loadParams(const int runNumber = 20076002)
+		{
+			
+			const int yearNumber = runNumber/1000000 - 1 + 2000;
+			const int date       = yearNumber*1000*10 + 101; // year + 01-01(Date)
 
-		if (!DB) {
-			LOG_WARN << "No data set found, creating new St_db_Maker... with date/time" << date << "/" << time << endm;
-			dbMk = new St_db_Maker("db", "MySQL:StarDb", "$STAR/StarDb");
+			St_db_Maker *dbMk = new St_db_Maker("db", "MySQL:StarDb", "$STAR/StarDb");
 			dbMk->SetDebug();
-			dbMk->SetDateTime(date,time); //! event or run start time, set to your liking
+			dbMk->SetDateTime(date,1);
+			cout<<"cout for test, data: "<<date<<endl;
 			dbMk->SetFlavor("ofl");
 			dbMk->Init();
 			dbMk->Make();
-		}
 
-		if (!DB) {
-			LOG_WARN << "Failed to connect to Database!" << endm;
-			return;
-		}
+			TDataSet *DB = 0;
+			DB = dbMk->GetDataBase("Calibrations/tof/tofSimResParams");
+
+			if(!DB)
+			{
+				LOG_WARN << "ERROR: could not read requested table from DB" << endm;
+				return;
+			}
 
-		St_tofSimResParams *dataset = 0;
-		dataset = (St_tofSimResParams*) DB->Find("tofSimResParams");
-
-		if (dataset) {
-			TDatime val[3];
-			dbMk->GetValidity((TTable*)dataset,val);
-			tofSimResParams_st* table = static_cast<tofSimResParams_st*>(dataset->GetTable());
-
-      mAverageTimeResTof=0;
-			for ( int i = 0; i < 120; i++ ){ //  nTrays
-				for ( int j = 0; j < 192; j++ ){
-					size_t index = i * 120 + j;
-					params[i][j] =  table[0].resolution[index];
-          mAverageTimeResTof+=params[i][j];
-					LOG_DEBUG << "tray:" << i << ", mod cell:" << j << " = " << table[0].resolution[index]  << " == " << params[i][j] << endm;
+			St_tofSimResParams *dataset = 0;
+			dataset = (St_tofSimResParams*) DB->Find("tofSimResParams");
+
+
+			if(dataset) 
+			{
+				TDatime val[2];
+				dbMk->GetValidity((TTable*)dataset,val);
+				
+				std::cout << "Dataset validity range: [ " << val[0].GetDate() << "." << val[0].GetTime() << " - "
+					<< val[1].GetDate() << "." << val[1].GetTime() << " ] " << std::endl;
+				
+				tofSimResParams_st* tofTable = static_cast<tofSimResParams_st*>(dataset->GetTable());
+				
+				mAverageTimeResTof = 0;
+				for ( int i = 0; i < 120; i++ )
+				{
+					for ( int j = 0; j < 192; j++ )
+					{
+						size_t index = i * 192 + j;
+						params[i][j] = tofTable[0].resolution[index];
+
+						mAverageTimeResTof += params[i][j];
+
+						LOG_DEBUG << "tray:" << i << ", mod cell:" << j << " = " << tofTable[0].resolution[index]  << " == " << params[i][j] << endm;
+						cout<<"index: "<<index<<endl;
+						LOG_INFO <<  "tray:" << i << ", mod cell:" << j << " = " << tofTable[0].resolution[index]  << " == " << params[i][j] << endm;
```

Comment:
Do these LOG_DEBUG and LOG_INFO print out exactly the same information?
Remove LOG_INFO. Remove cout

Reply:
I only ran LOG_INFO in previous test, LOG_INFO could print out the full table without problem.
LOG_INFO line was removed